alacritty/src/input.rs

```diff
@@ -1267,6 +1279,14 @@ mod tests {
         fn on_typing_start(&mut self) {
             unimplemented!();
         }
+
+        fn to_string(&self) -> String {
+            self.terminal.grid_to_string()
+        }
+
+        fn to_string_only_visible(&self) -> String {
+            self.terminal.grid_to_string_only_visible()
+        }
```

Comment:
Why make this a function on the grid and terminal, when we can just non-mutably access the grid from the terminal using `self.terminal.grid()`?

Reply:
It is implemented on the terminal and action context but not on the grid.

Reply:
Ah, it's not implemented on the grid because that doesn't know about its cell's contents, right?

Reply:
That is correct. It is implemented on `Term<T>` because the grid is always `Grid<Cell>`.